pkg/kvcache/kvevents/events.go

```diff
@@ -0,0 +1,79 @@
+package kvevents
+
+import (
+	"fmt"
+
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+// Event is a marker interface for KV cache events.
+type Event interface {
+	isEvent()
```

Comment:
`Event` interface is package private (unexported `isEvent` method). 
Consider if you might be able to make the interface private to begin with? 
```go
type event interface{ ... }
```